src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -482,32 +495,46 @@ StringLiteralEntry *GlobalStringLiteralMap::AddStringLiteral(EEStringData *pStri
 
     StringLiteralEntry *pRet;
 
+    bool isFrozen = false;
+    STRINGREF strObj = AllocateStringObject(pStringData, preferFrozenObjHeap, &isFrozen);
+    GCPROTECT_BEGIN(strObj)
```

Comment:
Nit: The `GC_PROTECT` is only needed on the `!isFrozen` path.

Reply:
Fixed, thanks!